ts/sagas/watchEmailValidatedChangedSaga.ts

```diff
@@ -0,0 +1,37 @@
+import { fromNullable, none, Option } from "fp-ts/lib/Option";
+import { Effect, put, takeEvery } from "redux-saga/effects";
+import { ActionType, getType } from "typesafe-actions";
+import { emailValidationChanged } from "../store/actions/emailValidationChange";
+import { profileLoadSuccess } from "../store/actions/profile";
+
+// tslint:disable-next-line:no-let
+let maybePreviousEmailValidated: Option<boolean> = none;
+
+/**
+ * This saga check for the event of `profileLoadSuccess` and check when the email validation is completed
+ * in order to dispatch the event to mixpanel
+ * @param initialEmailValidated: the initial email validate value from profile, before refresh
+ */
+export function* watchEmailValidatedChangedSaga(
+  initialEmailValidated: Option<boolean>
+): IterableIterator<Effect> {
+  maybePreviousEmailValidated = initialEmailValidated;
+  yield takeEvery(getType(profileLoadSuccess), checkEmailChanged);
+}
+
+function* checkEmailChanged(action: ActionType<typeof profileLoadSuccess>) {
+  const profileUpdate = action.payload;
+
+  // dispatch the action only if a previous state exists.
+  const emailStateChanged = maybePreviousEmailValidated
```

Comment:
I'm not sure about what this PR should do:
> Short description:
When the email is validated, send the event to mixpanel.

I'm quite sure this code produces **_emailValidationChanged_** on every **is_email_validated** changes

- previous **true** actual **false** -> `emailStateChanged === true`
- previous **false** actual **true** -> `emailStateChanged === true` (this is the case we are looking for)
- previous **false/true** actual **undefined** -> `emailStateChanged === true`

So if we want to dispatch an event saying: "the user email has been validated" the code could be something like follows

```typescript
function* checkEmailValidated(
  action: ActionType<typeof profileLoadSuccess>
) {
  const profileUpdate = action.payload;
  // do nothing
  if (profileUpdate.is_email_validated !== true) {
    maybePreviousEmailValidated = fromNullable(
      profileUpdate.is_email_validated
    );
    return;
  }

  // dispatch the action only if a previous state exists and the previous value is false (false -> true)
  const hasBeenEmailValidated = maybePreviousEmailValidated
    .map(x => x === false)
    .getOrElse(false);

  if (hasBeenEmailValidated) {
    yield put(
     // we don't need a boolean if in the analytics we consider only the true case
    // instead launch the event only when the condition is satisfied
      profileEmailValidated()
    );
  }

  maybePreviousEmailValidated = fromNullable(profileUpdate.is_email_validated);
}
```

Further more the action **emailValidationChanged** could be **emailValidated**

Reply:
The idea here was to have an unique action (`emailValidationChanged`) that represents the change of state of `is_email_validated` with the payload containing the new state, instead of an event that only says "The mail is validated". 
In this way, the event of changing `is_email_validated` is decoupled from the action to send the event to mixpanel (only when the payload is `true`) and can be reused in the future if is needed to detect also when someone choose to change the email (true->false). 
If this seems too confusing I can revert to my first implementation as a single action `emailValidated` dispatched only when change from (false->true).

Reply:
Yes, it makes sense.
Anyway when we read the user time line is hard to understand what is happening because we have to read the payload value.
What do you think to make 2 differents action describing each case:

- PROFILE_EMAIL_VALIDATED
- PROFILE_EMAIL_CHANGED

In this way we can have a clean idea what user did reading his timeline.
Does it make sense?

Reply:
At the moment no payload is sent to mixpanel, only the event `EMAIL_VALIDATION_COMPLETED`.
In the current implementation we have an action, internal to the application, named `emailValidationChanged` that have a payload with the information of the new value of the `email_validated`. 
The meaning within the application context is that the value of the `email_validated` is changed and the new value is the one included in the payload.

This event is intercepted by the analytics middielware and when the payload is true (the mail is changed -> new value true -> meaning that the validation is completed), it sends to mixpanel the event `EMAIL_VALIDATION_COMPLETED`with a transformation of meaning.

```typescript
const emailValidationComplete = "EMAIL_VALIDATION_COMPLETED";
[...]
    case getType(emailValidationChanged):
      if (action.payload) {
        return mp.track(emailValidationComplete);
      }
```
To sum up:
| Action        | Payload           | Meaning  |
| ------------- |:-------------:| -----:|
| emailValidationChanged      | false | the profile email validation value is changed, was true now false |
| emailValidationChanged      | true      |   the profile email validation value is changed, was false now true |

If it seems confusing I could change with two separate actions also within the application.